watson-developer-cloud/discovery-components
# feat: structured query add and remove rules

#### What do these changes do/fix?
This adds in the ability to add and remove rule rows from rule rows in the `StructuredQuery` component. When clicking the `Add rule` button, you should see a new row of rules added. When there is more than one rule row, a `Remove row` icon should appear on the right side of the rule row. When that is clicked, the row should be removed. I set up a max number of allowed sibling rule rows, which is set to 3 for now, pending user feedback.

related to issue: https://github.ibm.com/watson-discovery/disco-widgets/issues/182

#### How do you test/verify these changes?
I added some tests for these new additions, so you can look over those and confirm they are fully and correctly testing these new features. You can also run Storybook and add/remove rules in the `StructuredQuery` default story there.

#### Have you documented your changes (if necessary)?
`StructuredQuery` is in Storybook, but nothing more was added there specifically for these features.

#### Are there any breaking changes included in this pull request?
no
